src/biz/tugay/groovyship/cli/CliGameController.groovy

```diff
@@ -0,0 +1,107 @@
+package biz.tugay.groovyship.cli
+
+import biz.tugay.groovyship.modal.Board
+import biz.tugay.groovyship.service.GameService
+
+import java.nio.charset.StandardCharsets
+
+import static java.lang.Integer.parseInt
+
+class CliGameController
+{
+
+  static GameService gameService = new GameService()
+
+  static Scanner scanner = new Scanner(System.in, StandardCharsets.UTF_8.toString())
+
+  static BoardCommandLinePrinter boardCommandLinePrinter = new BoardCommandLinePrinter()
+
+  static Board board = null
+
+  static void main(String[] args) {
+    println "Type ng to create a new game at any time or exit to stop playing."
+    println "Send missile at coordinates in the format: columnIndex,rowIndex."
+    println "Top left corner is coordinates 0,0"
+    println "Type ng to start a new game now."
+
+    while (true) {
+      if (board) {
+        boardCommandLinePrinter.print board
+        if (gameService.allShipsSank board) {
+          println "Congratulations, you win. Your game has ended. "
+          board = null
+        }
+      }
+
+      if (!board) {
+        println "Enter one of: ng | exit"
+      }
+      else {
+        println "Enter coordinates: 'column,row' to send a missile, 'ng' for a new game, 'exit' to exit."
+      }
+
+      def userInput = getUserInput()
+      if ("exit" == userInput) {
```

Comment:
`==` in Groovy is the `equals` method in Java.

Reply:
same with python, I always still double check though. It gives me a bad feeling when I see == with java.util.String